bin/sozo/src/utils.rs

```diff
@@ -0,0 +1,33 @@
+use anyhow::Result;
+use dojo_world::utils::{execution_status_from_maybe_pending_receipt, TransactionWaiter};
+use starknet::core::types::{ExecutionResult, InvokeTransactionResult};
+use starknet::providers::Provider;
+
+pub async fn handle_transaction_result<P>(
+    provider: P,
+    transaction_result: InvokeTransactionResult,
+    wait_for_tx: bool,
+    show_receipt: bool,
+) -> Result<()>
+where
+    P: Provider + Send,
+{
+    if wait_for_tx {
+        let receipt =
+            TransactionWaiter::new(transaction_result.transaction_hash, &provider).await?;
+
+        if show_receipt {
+            println!("{}", serde_json::to_string_pretty(&receipt)?);
+        } else {
+            let status = match execution_status_from_maybe_pending_receipt(&receipt) {
+                ExecutionResult::Succeeded => "Transaction OK",
+                ExecutionResult::Reverted { reason: _ } => "Transaction REVERTED",
+            };
+            println!("{status}");
```

Comment:
We may want the hash to always be exposed to the user, along with the status in this specific case.